spec/enforce_encoding_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'spec_helper'
+
+describe 'enforce encoding' do
+  before :all do
+    @without_multibyte = <<~HL7.gsub("\n", "\r").chop
```

Comment:
To make the test easier readable you could use the same name in both and then just replace the string in one of them with 'Joaqu\xEDn' this way it will be easier to read it later.

like this: 
```ruby
@without_multibyte = <<~HL7.gsub("\n", "\r").chop
...
     |jdelon^DELON^Joaquín|
...
    HL7

@with_multibyte = @without_multibyte.sub! 'Joaquín', 'Joaqu\xEDn'
```

I didn't test it but it should work.

Reply:
Done with minor modifications